extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -79,17 +79,40 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     @Override
     public void initialize() {
         GatewayConfig configuration = getConfigAs(GatewayConfig.class);
+
+        if (isConfigurationValid(configuration)) {
+            scheduler.schedule(new Runnable() {
+
+                @Override
+                public void run() {
+                    try {
+                        setupGateway(configuration);
```

Comment:
Did you analyze WHY this can take longer than 5 seconds?
After all, the code only initialises some Java instances where I do not expect any long running operations. The remote calls are already done in a separately scheduled job (`startScan`).

Reply:
I did not perform deeper analysis, since I could not reproduce initialization taking more than 5 seconds (all my trials resulted in the process taking no more than a second). However, since you said in the ticket that you tested on a HB in a weird state, I implemented a mechanism to prevent possible stall and make sure initialization always passes within the 5-second interval.

Reply:
I would avoid introducing async scheduled jobs just because of some guess. Looking at the log of #3685, it might even have hung in the configuration checks, so I am not sure if this code change here would at all address the issue. I would therefore rather refrain from this change.

Reply:
I understand your opinion, but what do you think about moving the configuration check plus gateway setup in the scheduled job altogether - in such case it is going to make sense and prevent from initialization slow-down. Furthermore, I am not that sure we can reproduce this whatsoever. :)

Reply:
>  moving the configuration check plus gateway setup in the scheduled job altogether

No, this does not make sense, because the framework already calls `initialize()` in a separate thread. Note that the 5s warning is really just a warning and the system fully recovers from it and continues - so instead of hiding the message, we should rather keep it as a mechanism to hint us at potential problems.

Reply:
Okay, in such a case, I also believe this is to be reverted. I will do that shortly.